### ISSUE TITLE
Heatmaps Do not look as expexted????

### ISSUE THREAD

**Issue by mkhoshle** [Arthur151/ROMP#316]
Hello,

I have trained ROMP on the `pw3d` dataset for 150 epochs and from SCRATCH. Here are my hyperparameters:
```
ARGS:
 tab: 'V1_hrnet' 
 dataset: 'pw3d'
 GPUS: 0,1,2,3
 distributed_training: False
 model_version: 1
 match_preds_to_gts_for_supervision: True

 master_batch_size: -1
 val_batch_size: 16
 batch_size: 16
 epoch: 150
 nw: 4
 nw_eval: 2
 lr: 0.00005
 exp: 'ROMP'              # experiment name
 save_dir: './results'           # Path to results directory

 fine_tune: True
 fix_backbone_training_scratch: False
 eval: True
 supervise_global_rot: False

 model_return_loss: True
 collision_aware_centermap: True
 collision_factor: 0.2
 homogenize_pose_space: True
 shuffle_crop_mode: True
 shuffle_crop_ratio_2d: 0.1
 shuffle_crop_ratio_3d: 0.4

 merge_smpl_camera_head: False
 head_block_num: 2

 backbone: 'hrnet'
 centermap_size: 64
 centermap_conf_thresh: 0.2

 model_path: None

loss_weight:
 MPJPE: 200.
 PAMPJPE: 360.
 P_KP2D: 400.
 Pose: 80.
 Shape: 6.
 Prior: 1.6
 CenterMap: 160.

sample_prob:
 pw3d: 1
```
The model seems to be well converged and here is the loss function:
<img width="433" alt="Screen Shot 2022-07-19 at 5 29 03 PM" src="https://user-images.githubusercontent.com/21323137/179852481-90262d53-0205-4366-a2b2-d46a351bdeb6.png">

However, when I look at the Center-map heatmaps it doe
<img width="1033" alt="Screen Shot 2022-07-19 at 5 27 16 PM" src="https://user-images.githubusercontent.com/21323137/179852604-71ee525b-97ed-4dff-93df-2a3a73bbfd86.png">
s not look good at all:

I expected to get a plot similar to below that I obtained by fine-tuning the ROMP with HRNet-32 as the backbone (V1_hrnet_3dpwft.sh):
<img width="1234" alt="Screen Shot 2022-07-15 at 4 52 37 PM" src="https://user-images.githubusercontent.com/21323137/179852819-04c84b8b-8af6-4d50-8f29-76c0ef7be9f8.png">

Any thought or advice on why this is the case and what I further need to do to improve the center map heatmap?

Thanks in advance,

**Comment by Arthur151** [comment]
Hi, nice try.
But I am afraid that current training scheme (training on 3dpw only from scratch) would suffer from the generalization problem. 
Please note that ROMP has a pretraining stage that learns 2D pose estimation from rich in-the-wild 2D pose datasets.
The reason is that, as a one-stage method, we have to learn robust detection first, like other detection method, which is essential for the following representation learning. 
Training on 3dpw only from scratch would miss the process of learning robust detection, better generalization learned from in-the-wild datasets (like coco).
Because in 3dpw, not all people are annoted. Therefore, we don't use 3dpw to learn the heatmap detection, which means that in the whole training process, there is no supervision for the heatmap detection.

**Comment by Arthur151** [comment]
Hi, @mkhoshle 
After pretraining process, then we perform training initialized with the pretrained weights. In this process, we also need to use some 2D pose datasets to further improve the detection ability. 

In the pretraining stage, we have an additional branch to learn 2D pose estimation, of course, we also learn the center of the people in the image. 

About the pretrianing stage, I really recommand that you can directly use our pretrained checkpoints, or the pretrained backbone from 2D pose estimation methods, like Higher-HRNet for HRNet-32, simple-baseline for ResNet-50. They both work well. 

other video dataset:
indoor datasett, Human3.6M, MPI-INF-3DHP; in-the-wild Postrack

**Comment by mkhoshle** [comment]
Hi @Arthur151. Again, thank you very much for the clarification. In the paper, it is mentioned that the body center is calculated using a 2d pose. However, looking into figure 2a in the appendix it seems that there are two heads separately estimating body center and 2d pose. Which one is the case?

<img width="478" alt="Screen Shot 2022-07-22 at 3 41 00 PM" src="https://user-images.githubusercontent.com/21323137/180516531-a07d554b-f84d-4bca-91e1-428a977425b7.png">


1. Are the body center and 2d pose estimated separately or is the body center estimated using the 2d pose? 
2. If the body center is estimated using 2d pose, then why do you have two heads?

**Comment by Arthur151** [comment]
Hi, really sorry about that I missed this issue. 
    
  

 The 2D pose and body center heatmap are estimated by two individual heads. We need to supervise the estimated body center heatmap with the ground truth heatmap We calculate the ground truth body center location in heatmap using the ground truth 2d pose for supervising the estimated body center heatmap. The reason why we also use an individual head to estimate 2D pose heatmaps is that previous research (Taskonomy, best paper of cvpr18) testified that learning from related task is helpful for better representation learning. And we find that it is an important part to learn in the pretraining process.Thanks for your interests in ROMP.发件人: ***@***.*** ***@***.***> 代表 Mahzad Khoshlessan ***@***.***>发送时间: 星期六, 七月 30, 2022 6:35 上午收件人: Arthur151/ROMP ***@***.***>抄送: Yu Sun ***@***.***>; Mention ***@***.***>主题: Re: [Arthur151/ROMP] Heatmaps Do not look as expexted???? (Issue #316) 
@Arthur151 Hi. I would really appreciate it if you could help me with the above question.

**Comment by mkhoshle** [comment]
@Arthur151 Nice. Thanks so much.
